### DISCUSSION TITLE
Generating TXT files from Globals

### DISCUSSION THREAD

**Discussion by glitchbox42** [statamic/cms#7345] [Q&A]
I've been playing around with Statamic and was wondering if there's a way to generate txt files from a global set.

For example, the humans.txt or robots.txt files. I just can't seem to get it to work—I know, it's definitely me.

This is what I've got so far:

I created a 'robots.antlers.html' and placed it inside resources/views. I then added the following route to the web.php file under routes/.

'robots.antlers.html':
`{{ $content }}`

'web.php' file:
```
// The Robots.txt route to display the robots.txt file
Route::get('/robots.txt', function () {
    $content = config('humans_and_robots_txt.robots_txt_content');
    return view('robots', compact('content'));
});
```

**Comment by jasonvarga** [comment]
What you've done seems to work fine here. What's not working?

If you wanted to swap getting content from a config to a global, you can do this:

```php
$set = GlobalSet::find('settings')->inDefaultSite();
$content = $set->robots_content;
```

**Reply by glitchbox42** [reply]
For some reason, it's not actually outputting the content from the global field. The robots.txt will load, but it's always empty even though the Global has values in the textfield.

The only thing in the global textfield:
```
User-agent: *
Disallow:
```
